### PR TITLE
GEOPY-1840: Enforce reference data type shared between the tx_id_property of receiver and complement (transmitter, base station)

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-1840 - Enforce reference data type shared between the tx_id_property of receiver and complement (transmitter, base station)**

### CODE REVIEW

File: geoh5py/objects/surveys/electromagnetics/base.py [hunk 1]
```diff
@@ -466,6 +466,11 @@ def receivers(self, receivers: BaseEMSurvey):
         if receivers.tx_id_property is not None:
             self.edit_em_metadata({"Tx ID property": receivers.tx_id_property.uid})
 
+            if isinstance(
+                self.tx_id_property, ReferencedData | IntegerData
+            ) and isinstance(receivers.tx_id_property, ReferencedData | IntegerData):
```
Comment (MatthieuCMira) [inline geoh5py/objects/surveys/electromagnetics/base.py:471]: Question: why when IntegerData?

**Review by MatthieuCMira — APPROVED** [review]
Seems good for me. 
Can you just explain a bit why we need to passe the entity_type for IntegerData?
Thanks

File: geoh5py/objects/surveys/electromagnetics/base.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/objects/surveys/electromagnetics/base.py:471]: Same as for any data, if they share type, they share visual parameters.

Comment (MatthieuCMira) [reply geoh5py/objects/surveys/electromagnetics/base.py:471]: Then why resricting it to IntergerData and ReferencedData only?

Comment (domfournier) [reply geoh5py/objects/surveys/electromagnetics/base.py:471]: Enforced by GA

**Pull request merged**